tests/many_to_many/tests.py

```diff
@@ -555,3 +563,36 @@ def test_custom_default_manager_exists_count(self):
         self.assertEqual(
             self.p3.article_set.exists(), self.p3.article_set.all().exists()
         )
+
+class ManyToManyQueryTests(TestCase):
+    @classmethod
+    def setUpTestData(cls):
+        cls.article = Article.objects.create(headline='Django lets you build Web apps easily')
+        cls.nullable_target_article = NullableTargetArticle.objects.create(headline='The python is good')
+        NullablePublicationThrough.objects.create(article=cls.nullable_target_article, publication=None)
+
+    @skipUnlessDBFeature('supports_foreign_keys')
+    def test_count_join_optimization(self):
+        with CaptureQueriesContext(connection) as query:
```

Comment:
Use `self.assertNumQueries(1)` instead to highlight why these tests are wrongly passing

```suggestion
        with self.assertNumQueries(1) as queries:
```